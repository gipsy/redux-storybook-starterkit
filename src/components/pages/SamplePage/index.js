import React from 'react'

import PageTemplate from '~/src/components/templates/PageTemplate'
import Header from '~/src/components/organisms/Header'
import Footer from '~/src/components/organisms/Footer'
import Heading from '~/src/components/atoms/Heading'
import ReduxField from '~/src/components/organisms/ReduxField'
import Button from '~/src/components/atoms/Button'

const SamplePage = () => {
  return (
    <PageTemplate header={<Header />}>
      <Heading>Where is the company based?</Heading>
      <ReduxField input={{
          name: 'name',
          placeholder: 'enter you text here',
        }}
        meta={{}}
      />
      <Button text="Next" onClick={() => { alert('Test!'); }} />
    </PageTemplate>
  )
}

export default SamplePage
