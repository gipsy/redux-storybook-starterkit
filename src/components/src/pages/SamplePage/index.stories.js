import React from 'react'
import { storiesOf } from '@storybook/react'
import { SamplePage } from 'components/src'

storiesOf('Pages', module)
  .add('SamplePage', () => (
    <SamplePage />
  ))
  .disable(enableFn => {false})
