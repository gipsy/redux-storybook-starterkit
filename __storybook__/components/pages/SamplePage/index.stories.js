import React from 'react'
import { storiesOf } from '@storybook/react'
import { SamplePage } from '~/src/components/pages/SamplePage'

storiesOf('Pages', module)
  .add('SamplePage', () => (
    <SamplePage />
  ))
  .disable(enableFn => {false})
