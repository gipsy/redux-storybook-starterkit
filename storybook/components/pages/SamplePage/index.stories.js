import React from 'react'
import { storiesOf } from '@storybook/react'
import SamplePage from '~/src/components/pages/SamplePage'

storiesOf('Components/Pages', module)
  .add('SamplePage', () => (
    <SamplePage />
  ))
