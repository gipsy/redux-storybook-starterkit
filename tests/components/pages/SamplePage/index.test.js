import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import SamplePage from '~/src/components/pages/SamplePage'

it('renders a snapshot', () => {
  const tree = renderer.create(<SamplePage header="header" footer="footer"/>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders', () => {
  shallow(<SamplePage header="header" footer="footer" />)
})
