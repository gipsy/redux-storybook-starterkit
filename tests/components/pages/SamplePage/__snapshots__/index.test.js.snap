// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="sc-bdVaJa kyldDh"
>
  <header
    className="sc-bwzfXH fkWUxN"
  >
    <div
      className="sc-gqjmRU fObOYL sc-dnqmqq cTuhVT"
    >
      <div
        className="sc-VigVT gcLekX"
      >
        <span
          className="sc-iwsKbI eXZgwI"
        >
          <span
            className="sc-gZMcBi bndDEq"
          />
          <span
            className="sc-gZMcBi bndDEq"
          />
          <span
            className="sc-gZMcBi bndDEq"
          />
        </span>
      </div>
    </div>
  </header>
  <section
    className="sc-htpNat eBEIfo"
  >
    <h1
      className="sc-cSHVUG brYHsq"
    >
      Where is the company based?
    </h1>
    <div
      className="sc-dxgOiQ gxwVJt"
    >
      <input
        aria-describedby="nameError"
        className="sc-kGXeez cIxeuZ"
        height={50}
        id="name"
        name="name"
        placeholder="enter you text here"
        type="text"
      />
    </div>
    <button
      className="sc-eNQAEJ cPaiDp"
      height={50}
      label="Next"
      onClick={[Function]}
      type="button"
    >
      Button Example
    </button>
  </section>
  <footer
    className="sc-bxivhb bQeSTw"
  />
</div>
`;
